core/shared/src/main/scala/org/http4s/HttpVersion.scala

```diff
@@ -64,10 +68,51 @@ final case class HttpVersion private (major: Int, minor: Int)
     */
   override def compare(that: HttpVersion): Int =
     (this.major, this.minor).compare((that.major, that.minor))
+
+  override def equals(o: Any): Boolean =
+    o match {
+      case that: HttpVersion =>
+        this.major == that.major && this.minor == that.minor
+      case _ => false
+    }
+
+  override def hashCode(): Int = {
+    var hash = HttpVersion.hashSeed
+    hash = MurmurHash3.mix(hash, major)
+    hash = MurmurHash3.mix(hash, minor)
+    MurmurHash3.finalizeHash(hash, 2)
+  }
+
+  def productArity: Int = 2
+
+  def productElement(n: Int): Any =
+    n match {
+      case 0 => major
+      case 1 => minor
+      case _ => throw new IndexOutOfBoundsException()
+    }
+
+  def canEqual(that: Any): Boolean =
+    that match {
+      case _: HttpVersion => true
+      case _ => false
+    }
 }
 
 object HttpVersion {
 
+  /** Unsafe constructor for [[HttpVersion]], consider
+    * using its safe version [[HttpVersion.fromVersion]] instead.
+    */
+  def apply(major: Int, minor: Int): HttpVersion = new HttpVersion(major, minor)
```

Comment:
In 0.23 this constructor is available in public API, so I did the same here. But if we want, we can easily hide it from the public. Thoughts? cc @rossabaker

Reply:
I think, since this PR targets `main` and we don't worry about bin-compat issues in there, such a method just should not exist. Because it makes `HttpVersion` unsafe regardless of a bunch of its `from*` methods.

Reply:
..however, we could consider adding an `unsafeFromVersion` method which would resemble `apply` but apparently would make the fact it is unsafe more sounding.